packages/controller/lib/setup/setupRepo.js

```
@@ -227,154 +215,115 @@ function Repo(options) {
+            if (repoObj) {
+                if (!repoObj.native.repositories[repoName]) {
+                    return `Repository "${repoName}" not found.`;
                 } else {
-                    objects.getObject('system.repositories', (err, obj) => {
-                        if (err) {
-                            callback && callback(err);
-                        } else if (!obj) {
-                            callback && callback();
-                        } else {
-                            if (!obj.native.repositories[repoName]) {
-                                callback && callback(`Repository "${repoName}" not found.`);
-                            } else {
-                                delete obj.native.repositories[repoName];
-                                obj.from = `system.host.${tools.getHostName()}.cli`;
-                                obj.ts = Date.now();
-                                objects.setObject('system.repositories', obj, callback);
-                            }
-                        }
-                    });
+                    delete repoObj.native.repositories[repoName];
+                    repoObj.from = `system.host.${tools.getHostName()}.cli`;
+                    repoObj.ts = Date.now();
+                    await objects.setObjectAsync('system.repositories', repoObj);
                 }
             }
-        });
+        }
     };
 
-    this.setActive = function (repoName, callback) {
-        objects.getObject('system.repositories', (err, obj) => {
-            if (err) {
-                return callback && callback(err);
-            } else {
-                obj = obj || defaultSystemRepo;
+    this.setActive = async function (repoName) {
+        const obj = (await objects.getObjectAsync('system.repositories')) || defaultSystemRepo;
+        if (!obj.native.repositories[repoName]) {
+            return `Repository "${repoName}" not found.`;
```

error cb would let it throw